File: clojure/perfect-numbers/src/perfect_numbers.clj
(ns perfect-numbers)

(defn divisor-sum [n]
  (apply + (filter #(= 0 (mod n %)) (range 1 n))))

(defn classify [n]
  (if (> 0 n)
    (throw (IllegalArgumentException. ""))
    (let [ds (divisor-sum n)]
      (cond
        (> n ds) :deficient
        (< n ds) :abundant
        :else :perfect))))

File: csharp/perfect-numbers/PerfectNumbers.cs
using System;
using System.Linq;

public enum Classification
{
    Perfect,
    Abundant,
    Deficient
}

public static class PerfectNumbers
{
    public static Classification Classify(int number)
    {
        var sumDivisors = (from i in Enumerable.Range(1, number - 1) where number % i == 0 select i).Sum();
        if (number == sumDivisors)
        {
            return Classification.Perfect;
        }
        else if (number < sumDivisors)
        {
            return Classification.Abundant;
        }
        return Classification.Deficient;
        
    }
}
